src/lavinmq/amqp/queue/priority_queue.cr

```diff
@@ -7,52 +7,185 @@ module LavinMQ::AMQP
       @effective_args << "x-max-priority"
     end
 
-    private def init_msg_store(data_dir)
+    private def init_msg_store(msg_dir)
       replicator = durable? ? @vhost.@replicator : nil
-      PriorityMessageStore.new(data_dir, replicator, metadata: @metadata)
+      max_priority = @arguments["x-max-priority"]?.try(&.as?(Int)) || 0u8
+      PriorityMessageStore.new(max_priority.to_u8, msg_dir, replicator, metadata: @metadata)
     end
 
     class PriorityMessageStore < MessageStore
-      # def initialize(@data_dir : String, metadata : ::Log::Metadata, @replicator : Clustering::Replicator?)
-      def initialize(*args, **kwargs)
-        super
-        order_messages
+      # These are just to make the compiler happy. They are never used.
+      @acks = uninitialized Hash(UInt32, MFile)
+      @wfile = uninitialized MFile
+      @rfile = uninitialized MFile
+      @wfile_id = 0
+      @rfile_id = 0
+
+      def initialize(
+        @max_priority : UInt8,
+        @msg_dir : String,
+        @replicator : Clustering::Replicator?,
+        @durable : Bool = true,
+        @metadata : ::Log::Metadata = ::Log::Metadata.empty,
+      )
+        @log = Logger.new(Log, metadata.extend({max_prio: @max_priority.to_s}))
+        @stores = Array(MessageStore).new(1 + @max_priority)
+
+        # Setup all sub-stores. We want the highets priority store
+        # first in the array since we'll always search from high to
+        # low priority.
+        (0..@max_priority).reverse_each do |i|
+          sub_msg_dir = File.join(@msg_dir, i.to_s)
+          Dir.mkdir_p sub_msg_dir
```

Comment:
Maybe calls for an `init` or `setup` method? Creating folders in the constructor seems a bit too much of a side effect?